src/multiplexed-session.ts

```diff
@@ -0,0 +1,241 @@
+
+  /**
+   * Creates a new multiplexed session and manages its maintenance.
+   *
+   * This method initiates the session creation process by calling the `_createSession` method, which returns a Promise.
+   */
+  createSession(): void {
+    this._createSession()
+      .then(() => {
+        this._maintain();
+      })
+      .catch(err => {
+        if (
+          isDatabaseNotFoundError(err) ||
+          isInstanceNotFoundError(err) ||
+          isCreateSessionPermissionError(err) ||
+          isDefaultCredentialsNotSetError(err) ||
+          isProjectIdNotSetInEnvironmentError(err)
+        ) {
+          return;
+        }
+        this.emit('error', err);
+      });
+  }
+
+  /**
+   * Creates a new multiplexed session.
+   *
+   * This method sends a request to the database to create a new session with multiplexing enabled.
+   * The response from the database would be an array, the first value of the array will be containing the multiplexed session.
+   *
+   * @returns {Promise<void>} A Promise that resolves when the session has been successfully created and assigned, an event
+   * `mux-session-available` will be emitted to signal that the session is ready.
+   *
+   * In case of error, an error will get emitted along with the erorr event.
+   *
+   * @private
+   */
+  async _createSession(): Promise<void> {
+    try {
+      const [createSessionResponse] = await this.database.createSession({
+        multiplexed: true,
+      });
+      this._multiplexedSession = createSessionResponse;
+      this.emit('mux-session-available');
+    } catch (e) {
+      this.emit('error', e);
+    }
+  }
+
+  /**
+   * Maintains the multiplexed session by periodically refreshing it.
+   *
+   * This method sets up a periodic refresh interval for maintaining the session. The interval duration
+   * is determined by the @param refreshRate option, which is provided in days.
+   * The default value is 7 days.
+   *
+   * @returns {void} This method does not return any value.
+   *
+   * @throws {Error} If the `refreshRate` option is not defined, this method could throw an error.
+   */
+  _maintain(): void {
+    const refreshRate = this.refreshRate! * 24 * 60 * 60000;
+    this._refreshHandle = setInterval(() => this._refresh(), refreshRate);
+    this._refreshHandle.unref();
+  }
+
+  /**
+   * Refreshes the session by checking its expiration time and recreating the session if expired.
+   * Every 7th day a new Multiplexed Session will get created. However, a Multiplexed Session will
+   * be alive for 30 days in the backend.
+   *
+   * @returns {Promise<void>} A Promise that resolves once the refresh process is completed.
+   *                          If the session is expired, a new session will be created.
+   *
+   * @throws {Error} If there is an issue with retrieving the session metadata or calculating the expiration time.
+   */
+  async _refresh(): Promise<void> {
+    const metadata = await this._multiplexedSession?.getMetadata();
```

Comment:
Yes we can store it in multiplexed session class.
In Java we store the expiration time of session, instead of creation time to simplify checks
https://github.com/googleapis/java-spanner/blob/b545557b1a27868aeb5115b3947d42db015cc00e/google-cloud-spanner/src/main/java/com/google/cloud/spanner/MultiplexedSessionDatabaseClient.java#L183